regression/apps/file_io/file_io.c

```diff
@@ -40,99 +180,94 @@ long calc_duration(struct timespec *start, struct timespec *end)
 	       (end->tv_nsec - start->tv_nsec);
 }
 
-int perform_sequential_io(int fd, ssize_t (*io_func)(int, void *, size_t),
-			  const char *op_name)
+int perform_io_bench(int fd, size_t file_size, size_t buffer_size,
+		     ssize_t (*io_func)(int, void *, size_t),
+		     const char *op_name, const char *io_skew)
 {
 	struct timespec start, end;
-	char buffer[BUFFER_SIZE];
-	ssize_t ret, offset = 0;
 	long total_nanoseconds = 0, avg_latency;
 	double throughput;
+	ssize_t ret_bytes = 0;
+	char *buffer;
+	if (posix_memalign((void **)&buffer, BLOCK_SIZE, buffer_size) != 0) {
+		fprintf(stderr, "Failed to allocate memory.\n");
+		return -1;
+	}
 
-	memset(buffer, 0, BUFFER_SIZE);
+	memset(buffer, 0, buffer_size);
 	lseek(fd, 0, SEEK_SET);
 
+	clock_gettime(CLOCK_MONOTONIC, &start);
 	for (int i = 0; i < NUM_OF_CALLS; i++) {
-		if (offset >= FILE_SIZE) {
-			offset = lseek(fd, 0, SEEK_SET);
+		if (strcmp(io_skew, "sequential") == 0) {
+			off_t cur_offset = lseek(fd, 0, SEEK_CUR);
+			if (cur_offset >= file_size) {
+				lseek(fd, 0, SEEK_SET);
+			}
+		} else {
+			off_t random_offset =
+				(random() % (file_size / buffer_size)) *
+				buffer_size;
+			lseek(fd, random_offset, SEEK_SET);
+		}
```

Comment:
The performance may be negatively impacted by the use of the `lseek` operation. So I recommend not to include the time of `lseek`.

Reply:
Same as the above comment to mimic a complete sequential/random workload. By the way, if the pread/pwriteis ready, we can get rid of the `lseek`'s offset lock cost.